great_expectations/experimental/metric_repository/column_descriptive_metrics_metric_retriever.py

```diff
@@ -25,145 +28,269 @@ class ColumnDescriptiveMetricsMetricRetriever(MetricRetriever):
+                )
+                metrics.append(
+                    ColumnMetric[float](
+                        batch_id=batch_id,
                         metric_name=metric_name,
-                        metric_domain_kwargs={"column": column},
-                        metric_value_kwargs={},
+                        column=column,
+                        value=value,
+                        exception=exception,
                     )
                 )
-        validator = self._context.get_validator(batch_request=batch_request)
-        computed_metrics = validator.compute_metrics(
-            metric_configurations=column_metric_configs,
-            runtime_configuration={"catch_exceptions": True},  # TODO: Is this needed?
+
+        return metrics
+
+    def _get_non_numeric_column_metrics(
+        self, batch_request: BatchRequest, column_list: List[str]
+    ) -> Sequence[Metric]:
+        column_metric_names = [
+            "column_values.null.count",
+        ]
+
+        column_metric_configs = self._generate_column_metric_configurations(
+            column_list, column_metric_names
         )
+        batch_id, computed_metrics = self._compute_metrics(
+            batch_request, column_metric_configs
+        )
+
         # Convert computed_metrics
         metrics: list[Metric] = []
         ColumnMetric.update_forward_refs()
+        metric_lookup_key: _MetricKey
+
+        for metric_name in column_metric_names:
+            for column in column_list:
+                metric_lookup_key = (metric_name, f"column={column}", tuple())
+                value, exception = self._get_metric_from_computed_metrics(
+                    metric_name=metric_name,
+                    metric_lookup_key=metric_lookup_key,
+                    computed_metrics=computed_metrics,
+                )
+                metrics.append(
+                    ColumnMetric[int](
+                        batch_id=batch_id,
+                        metric_name=metric_name,
+                        column=column,
+                        value=value,
+                        exception=exception,
+                    )
+                )
+
+        return metrics
 
+    def _get_all_column_names(self, metrics: Sequence[Metric]) -> List[str]:
+        column_list: List[str] = []
+        for metric in metrics:
+            if metric.metric_name == "table.columns":
+                column_list = metric.value
+        return column_list
+
+    def _get_numeric_column_names(self, batch_request: BatchRequest) -> list[str]:
+        """Get the names of all numeric columns in the batch."""
+        validator = self._context.get_validator(batch_request=batch_request)
+        domain_builder = ColumnDomainBuilder(
+            include_semantic_types=[SemanticDomainTypes.NUMERIC],
+        )
         assert isinstance(validator.active_batch, Batch)
         if not isinstance(validator.active_batch, Batch):
```

Comment:
We may never get here if we hit the assertion error in the previous line.

Reply:
I think the error message is helpful though.

Reply:
todo: move the error message to the `assert` statement and change it into an `AssertionError`